### ISSUE TITLE
Best practices for perl libraries with lots of dependencies?

### ISSUE THREAD

**Issue by jerowe** [bioconda/bioconda-recipes#1438]
Are there any best practices for this kind of thing?

For instance, I see in the perl-bio-db-sam, the build script is a call to cpanm. That's peachy by me, but if you'd prefer I build skeleton's for each of the dependencies and a pull request for each I could do that too.

**Comment by daler** [comment]
I think a skeleton for each dependency would be preferable so that each can be installed separately if needed.

Luckily you don't have to submit a pull request for each -- the testing and building setup will handle multiple recipes at once.

**Comment by johanneskoester** [comment]
I agree. At some point, we could add recursive skeleton capabilities to bioconda-utils.

**Comment by jerowe** [comment]
An update for this - the conda skeleton cpan does have a recursive option, but it chokes on Test::Deep. I opened an issue with the conda-build.

The only problem with this approach is that there is a mix of requiring perl and perl-threaded, and I've been told having both in a single env is problematic. ;-)

So far I am solving this issue with sed to make sure all the dependencies require perl-threaded and not perl, ( I did this for the deps of [https://github.com/bioconda/bioconda-recipes/tree/master/recipes/augustus/3.2.2](Augustus-3.2.2) ), but this could potentially be error prone.

Ideas, thoughts, complaints?
